qiskit/pulse/schedule.py

```diff
@@ -790,36 +791,15 @@ def __eq__(self, other: Union['Schedule', Instruction]) -> bool:
 
         Equality is checked by verifying there is an equal instruction at every time
         in ``other`` for every instruction in this ``Schedule``.
-
-        .. warning::
-
-            This does not check for logical equivalency. Ie.,
-
-            ```python
-            >>> (Delay(10)(DriveChannel(0)) + Delay(10)(DriveChannel(0)) ==
-                 Delay(20)(DriveChannel(0)))
-            False
-            ```
         """
-        channels = set(self.channels)
-        other_channels = set(other.channels)
-
-        # first check channels are the same
-        if channels != other_channels:
-            return False
-
-        # then verify same number of instructions in each
-        instructions = self.instructions
-        other_instructions = other.instructions
-        if len(instructions) != len(other_instructions):
-            return False
-
-        # finally check each instruction in `other` is in this schedule
-        for idx, inst in enumerate(other_instructions):
-            # check assumes `Schedule.instructions` is sorted consistently
-            if instructions[idx] != inst:
+        # Remove non-functional operations from the schedule such as delays and zero phaseshifts 
+        first_schedule = _reduce_to_stack(self)
+        second_schedule = _reduce_to_stack(other)
+        
+        # Check that if the two new schedules are equal
+        for idx, inst in enumerate(second_schedule):
+            if first_schedule[idx] != inst:
```

Comment:
Reduce stack is not sufficient. The order of the instructions in two different identical schedules could be different, since there may be multiple instructions which start at the same time on different channels. I think this can be solved by iterating through the instructions on a per channel basis. The above checks that the channels are the same are necessary. Checking that the number of instructions is the same is also a good idea, because it can save some time.